File: frontend/src/types/cliente.ts
export interface Cliente {
  id: number;
  nombre: string;
  apellido: string;
  telefono: string;
  email: string;
  fechaRegistro: string;
  activo: boolean;
}

export interface CreateClienteDto {
  nombre: string;
  apellido: string;
  telefono: string;
  email: string;
}

export interface UpdateClienteDto {
  nombre: string;
  apellido: string;
  telefono: string;
  email: string;
}

File: GymApp/Models/Cliente.cs
using System.ComponentModel.DataAnnotations;

namespace GymApp.Models
{
    public class Cliente
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Nombre { get; set; }

        [Required]
        [MaxLength(100)]
        public string Apellido { get; set; }

        [MaxLength(20)]
        public string Telefono { get; set; }

        [EmailAddress]
        [MaxLength(150)]
        public string Email { get; set; }

        public DateTime FechaRegistro { get; set; } = DateTime.Now;

        public bool Activo { get; set; } = true;


        public List<Membresia> Membresias { get; set; } = new();
        public List<Pago> Pagos { get; set; } = new();
    }
}
